admin/docs/customizing_view_components.md

```diff
@@ -0,0 +1,190 @@
+  You can consider doing the same if you need to use one of its helpers.
+
+For example, the following replaces the main nav component:
+
+```ruby
+# app/components/my_application/solidus_admin/main_nav/component.rb
+class MyApplication::SolidusAdmin::MainNav::Component < SolidusAdmin::BaseComponent
+  # do your thing
+end
+```
+
+If you need more control, you can explicitly register your component in the
+Solidus Admin container instead of using an implicit path:
+
+> ⓘ  Right now, that will raise an error when the application is reloaded. We
+> need to fix it.
+
+```ruby
+# config/initalizers/solidus_admin.rb
+Rails.application.config.to_prepare do
+  SolidusAdmin::Container['components.main_nav.component'] = OtherNamespace::NavComponent
+end
+```
+
+### Tweaking a component
+
+If you only need to tweak a component, you can always inherit from it in a
+matching path from within your application (or manually register in the
+container) and override the methods you need to change:
+
+```ruby
+# app/components/my_application/solidus_admin/main_nav/component.rb
+class MyApplication::SolidusAdmin::MainNav::Component < ::SolidusAdmin::MainNav::Component
+  def sorted_items
+    super.reverse
+  end
+end
+```
+
+Be aware that this approach comes with an important trade-off: You'll need to
+keep your component in sync with the original one as it changes on future updates.
+For instance, in the example above, the component is overriding a private
+method, so there's no guarantee that it will continue to exist in the future
+without being deprecated, as we only guarantee public API stability.
+
+### Replacing or tweaking a nested component
+
+If you need to customize a nested component that is used in a single place, you
+can do that by following our previous instructions on [replacing a
+component](#replacing-a-component) or [tweaking a component](#tweaking-a-component).
+
+However, if the component you want to customize is used in multiple places, you
+can replace the default component called from the parent one.
+
+For the sake of example, let's say that the main navigation item component is used in
+multiple places, but you only need to change the one rendered from the main
+navigation. To begin with, you'd need to manually register in the container the
+component you want to use:
+
+```ruby
+Rails.application.config.to_prepare do
+  SolidusAdmin::Container.register('components.main_nav_item_foo.component', MyApplication::SolidusAdmin::MainNavItemFoo::Component)
+end
+```
+
+Then, you can change the default value of the `main_nav_item_component`
+argument from the top-level nav component:
+
+```ruby
+# app/components/my_application/solidus_admin/main_nav/component.rb
+class MyApplication::SolidusAdmin::MainNav::Component < ::SolidusAdmin::MainNav::Component
+  def initialize(main_nav_item_component: components('main_nav_item_foo'), **kwargs)
+    super
+  end
+end
+```
```

Comment:
I think this is a worthy feature to have, but I think we can achieve it tapping into the `components` helper, for example we might have an optional separate registry for each component and fallback to the global one.

What follow is pseudo-code:

```rb
# helper.rb
def components(name)
  container[name]
end

def container
  Container.for(self.class) || Container.global
end
```

```rb
# initializer in the host app
SolidusAdmin.configure do |config|
  config.component_registry_for('foo/bar')['baz'] = 'MyAlternativeBaz::Component'
end
```

Reply:
> We can try something and see how it looks!

👍 I can give it a go and se if things come out naturally 😌 or if it's a monster 👹